lib/timers.js

```diff
@@ -496,7 +501,15 @@ function rearm(timer) {
     timer._handle.start(timer._repeat);
   } else {
     timer._idleTimeout = timer._repeat;
-    active(timer);
+
+    const duration = TimerWrap.now() - start;
+    if (duration >= timer._repeat) {
+      // If callback duration >= timer._repeat,
+      // add 1 ms to avoid blocking eventloop
+      process.nextTick(insert, timer, false, start + duration - timer._repeat);
```

Comment:
I had meant replace the conditional with `nextTick` - since that doesn't seem to work (I forget why), we don't need it here.